spiketoolkit/comparison/comparisontools.py

```diff
@@ -0,0 +1,398 @@
+                matches = (np.abs(mapped_st.astype(int)-n_sp)<=delta_tp//2)
+                if np.sum(matches) > 0:
+                        lab_st1[sp_i] = 'TP'
+                        lab_st2[np.where(matches)[0][0]] = 'TP'
+        else:
+            # Sam garcia note : I guess this is a mistake
+            lab_st1 = np.array(['FN'] * len(sts1[u_i]))
+            # Sam note : I would add this line
+            #�labels_st1[u1] = lab_st1
+
+    # find CL-CLO-CLSO
+    for u_i, u1 in enumerate(sorting1.getUnitIds()):
+        lab_st1 = labels_st1[u1]
+        #~ st1 = sorting1.getUnitSpikeTrain(u1)
+        st1 = sts1[u_i]
+        for l_gt, lab in enumerate(lab_st1):
+            if lab == 'UNPAIRED':
+                for u_j, u2 in enumerate(sorting2.getUnitIds()):
+                    if u2 in unit_map12.values() and unit_map12[u1] != -1:
+                        lab_st2 = labels_st2[u2]
+                        st2 = sts2[u_j]
+
+                        n_up = st1[l_gt]
+                        id_sp = np.where((st2.astype(int)-n_sp)<=delta_tp//2)[0]
+                        if len(id_sp) > 0 and lab_st2[id_sp[0]] == 'UNPAIRED':
+                            lab_st1[l_gt] = 'CL_' + str(u1) + '_' + str(u2)
+                            lab_st2[id_sp[0]] = 'CL_' + str(u2) + '_' + str(u1)
+
+    for u1 in sorting1.getUnitIds():
+        lab_st1 = labels_st1[u1]
+        for l_gt, lab in enumerate(lab_st1):
+            if lab == 'UNPAIRED':
+                lab_st1[l_gt] = 'FN'
+
+    for u2 in sorting2.getUnitIds():
+        lab_st2 = labels_st2[u2]
+        for l_gt, lab in enumerate(lab_st2):
+            if lab == 'UNPAIRED':
+                lab_st2[l_gt] = 'FP'
+
+    TOT_ST1 = sum([len(sts1[i]) for i in range(len(sts1))])
+    TOT_ST2 = sum([len(sts2[i]) for i in range(len(sts2))])
+    total_spikes = TOT_ST1 + TOT_ST2
+    TP = sum([len(np.where('TP' == labels_st1[unit])[0]) for unit in sorting1.getUnitIds()])
+    CL = sum(
+        [len([i for i, v in enumerate(labels_st1[unit]) if 'CL' in v]) for unit in sorting1.getUnitIds()])
+    FN = sum([len(np.where('FN' == labels_st1[unit])[0]) for unit in sorting1.getUnitIds()])
+    FP = sum([len(np.where('FP' == labels_st2[unit])[0]) for unit in sorting2.getUnitIds()])
+    
+    counts = {'TP': TP, 'CL': CL, 'FN': FN, 'FP': FP, 'TOT': total_spikes, 'TOT_ST1': TOT_ST1,
+                   'TOT_ST2': TOT_ST2}
+
+    return counts, labels_st1, labels_st2
+    
+
+
+def do_confusion_matrix(sorting1, sorting2, unit_map12, labels_st1, labels_st2):
+    """
+    Compute the confusion matrix between two sorting.
+    
+    Parameters
+    ----------
+    sorting1: SortingExtractor instance
+        The ground truth sorting.
+    
+    sorting2: SortingExtractor instance
+        The tested sorting.
+
+    unit_map12: dict
+        Dict of matching from sorting1 to sorting2.
+        
+
+    Output
+    ----------
+    
+    confusion_matrix: the confusion matrix
+    
+    st1_idxs: ???
```

Comment:
st1_idxs: order of units1 in confusion matrix

st2_idxs: order of units2 in confusion matrix